doc/model.rst

```diff
@@ -0,0 +1,683 @@
+Models are flexible enough to support many different applications, however,
+to be used with SmartSim clients (SmartRedis) the application must be written
+in Python, C, C++, or Fortran.
+
+When initializing a ``Model``, users
+provide executable simulation code to the Models run settings as well as
+execution instructions with regard to the workload
+manager (e.g. Slurm) and available compute resources. Users can specify
+and assign values to parameters used within the simulation via the `params`
+initialization argument. Parameters supplied in the `params` argument can either be
+written into supplied configuration files or be assigned within configuration files
+for use within the simulation at runtime. This functionality is supported by the
+``Model.attach_generator_files()`` helper function users have access to
+once creating a Model instance.
+
+SmartSim supports **two** strategies for deploying Models:
+
+- **Standard Model**: Operating on a separate compute node from a
+  SmartSim orchestrator, Standard Models facilitate communication with a clustered
+  Orchestrator through SmartRedis clients. This architecture is ideal for simulations
+  leveraging distributed computing capabilities, ensuring parallelized and efficient
+  execution across multiple nodes.
+
+- **Colocated Model**: Sharing compute resources with a colocated Orchestrator on the
+  same compute node, Colocated Models offer advantages in scenarios where minimizing communication
+  latency is critical, such as online inference or runtime processing.
+
+SmartSim manages ``Model`` instances through the :ref:`Experiment API<experiment_api>` by providing functions to
+launch, monitor, and stop simulations. Additionally, Models can be launched individually
+or as a group via an Ensemble. Once a Model instance has been initialized, users have access to
+the :ref:`Model API<model_api>` helper functions.
+
+======================
+Initialize Model Types
+======================
+--------
+Overview
+--------
+
+The :ref:`Experiment API<experiment_api>` is responsible for initializing all workflow components.
+A ``Model`` is created using the ``Experiment.create_model()`` helper function. Users can customize the
+the Model by specifying initializer arguments.
+
+The key initializer arguments are:
+
+-  `name` (Type: str): Specify the name of the model for unique identification.
+-  `run_settings` (Type: RunSettings): Describe execution settings for a Model.
+-  `params` (dict, optional): Provides a dictionary of parameters for Models.
+-  `path` (str, optional): Path to where the model should be executed at runtime.
+-  `enable_key_prefixing` (bool, optional): Prefix the model name to data sent to the database to prevent key collisions. Default is True.
+-  `batch_settings` (BatchSettings | None): Describes settings for batch workload treatment.
+
+To initialize a ``Model``, a `name` and :ref:`RunSettings<settings-info>` reference is required.
```

Comment:
Nit: I have a personal pet peeve about commas in docs. I'd try to rephrase anything to make the statement more direct. 

Example: "A name and runsettings reference are required to intialize a model"

Reply:
I see and I agree